ADC/Core/Src/main.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * 	Implement a routine that reads the internal voltage reference (ADC_IN17)
+ * 	and internal temperature sensor (ADC_IN18).
+ * 	The internal reference voltage must be in millivolts and the temperature in Celsius.
+ *
+ * 	1 - Polling		[/]
+ * 	2 - Interrupt	[ ]
+ * 	3 - DMA			[ ]
+ */
+
+#include "stm32l053xx.h"
+#include "macros.h"
+#include "init.h"
+
+uint8_t		ADC_error	= 0;	//error management flag
+uint16_t	measure 	= 0,	//AD read after a conversion
+			v_ref 		= 0;	//internal reference voltage measured (after calculation)
+int16_t		temperature	= 0;	//temperature in Celsius degrees measured (after calculation)
+
```

Comment:
We use global vars in embedded systems only when it is extremely necessary. Please double check if those vars really need to be global

Reply:
Yes, my bad about global vars.

But, I'm getting this weird warning when I initialize them as local in main() - check the image attached. I only use v_ref there and, if I don't initialize v_ref, the compiler checks an error for that. If I initialize it as global, everything is good. The program works with the warning, by the way
![Screen Shot 2020-04-13 at 12 07 06](https://user-images.githubusercontent.com/61191550/79131893-bb01d700-7d7f-11ea-8d55-d1dbbdb0217b.png)

Reply:
tip: you can use `asm("nop");` command to place your break point instead of `while(1);`. That command makes the processor stops for one clock cycle.

Why does the compiler is giving you that warning?

Reply:
I have no idea. If I declare v_ref as a global variable, ok. If I declare it as a local variable inside main I got this error. If I don't declare it I got an error (obviously). In both cases (global or local) the code works. Can you guys notice something in that print?

Reply:
Please research about the problem and understand what is happening. As a tip: the solution is not making that var global.

Reply:
I've found someone with this warning, and somebody said this:
"Here you set a value to the "none" variable, but your compiler will tell you it's unused because you never test it with comparison operators, or you never pass it to a function."

In my code, if I do this:
_v_ref = (3000 * (int32_t) (*VREFINT_CAL)) / measure;	//*3000 so we have Vref in mV
v_ref = v_ref + 1;_
the warning goes away.

I've also found other cases that looked like the same as mine right here. What do you think?

Reply:
aaaaand, if I do this to "temperature":
_temperature = ((measure * VDD_APPLI / VDD_CALIB) - (int32_t) *TEMP30_CAL_ADDR);
//temperature = temperature * (int32_t)(130 - 30);
//temperature = temperature / (int32_t)(*TEMP130_CAL_ADDR - *TEMP30_CAL_ADDR);
//temperature = temperature + 30;_
it receives the same warning